### PR TITLE
Fix Linux build

### PR DESCRIPTION
**Pull request by maximsmol**
Minimal set of patches needed to build + run Linux builds and a bit of accompanying tooling.

This PR is to document that we need to merge the Linux changes into master once some of the blockers (commits marked "FIXME") are resolved. For the time being, you can merge `build-linux` into the latest `master` if you need to build on Linux.

**TODO:**
1) Re-setup `thirdparty/*`, the current version was produced by a hodgepodge of autotools commands and is in a funky state because of that.
2) Re-write `.gitignore` to follow autotools standards on what files need to be shipped (do not exclude `configure` and `Makefile` etc.)
3) Remove/hide NixOS tooling
4) Make sure `magic.sh` works on a clean install
5) Fix the official `steam_api.so` (currently crashes, caused by "not enough data" on an `CInterProcessPipe`)
6) Provide a release script similar to the one for the Windows version
7) Provide a force-install script that will re-copy all the build products used by the game. This is useful if the TF2 folder is copied over original build files since otherwise only a fully rebuild will help

### CODE REVIEW

**Comment by maximsmol** [comment]
Yeah so it technically runs through the use of superhacks, need to cleanup + push

File: magic.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  exec 3>&1 4>&2
+else
+  exec 3>&1 4>&2 &>/dev/null
+fi
+
+cflags='-D_GLIBCXX_USE_CXX11_ABI=0'
+configure_args=('--host=i686-unknown-linux-gnu' "CFLAGS=\"${cflags}\"" "CXXFLAGS=\"${cflags}\"" '--enable-shared=no')
+
+function makethirdparty() {
+  chmod u+x ./configure && ./configure "${configure_args[@]}" && make
```
Comment (powpingdone) [inline magic.sh:51]:
```suggestion
  aclocal && autoconf && automake -a && ./configure "${configure_args[@]}" && make
``` This made magic.sh wfm.

Comment (maximsmol) [reply magic.sh:51]: Yeah, `magic.sh` is currently experimental. I will fixup autotools soon

**Comment by int-72h** [comment]
With some minor edits, it nearly builds on Arch- VPC segfaults.
`devtools/bin/vpc: line 46: 181586 Segmentation fault      (core dumped) $SCRIPTPATH/$BINNAME $FORCEARG $@`

**Comment by powpingdone** [comment]
Ran into the same problem as well, I'm wondering if its a GCC issue/regression.

**Comment by melvyn2** [comment]
The bundled vpc does not work for everyone, which is why we don't bundle a prebuilt in #140 .

**Comment by powpingdone** [comment]
Ie, the vpc_linux that is built during magic.sh?

**Comment by melvyn2** [comment]
Not sure what you mean, but yes, vpc is built in build.sh (in #140).

**Comment by int-72h** [comment]
Does this work with #140 or is this completely separate?

**Comment by melvyn2** [comment]
> Does this work with #140 or is this completely separate?

#140 is based upon this branch.

**Comment by micwoj92** [comment]
> > Does this work with #140 or is this completely separate?
> 
> #140 is based upon this branch.

So first this needs be merged before #140 or it doesnt matter?

**Comment by melvyn2** [comment]
> > > Does this work with #140 or is this completely separate?
> > 
> > 
> > #140 is based upon this branch.
> 
> So first this needs be merged before #140 or it doesnt matter?

Either #140 needs to be merged first, because it merges into this pr's branch (NOT master), or this gets merged and 140 is rebased and retargeted onto master.

**Comment by micwoj92** [comment]
> > > > Does this work with #140 or is this completely separate?
> > > 
> > > 
> > > #140 is based upon this branch.
> > 
> > 
> > So first this needs be merged before #140 or it doesnt matter?
> 
> Either #140 needs to be merged first, because it merges into this pr's branch (NOT master), or this gets merged and 140 is rebased and retargeted onto master.

Ah I see, unfortunately this branch has conflicts. These must be resolved.

**Pull request closed without merging**